docs/misc/commands.md

```diff
@@ -0,0 +1,211 @@
+# Commands
+A command is a textual instruction for Minecraft to do somethings.
+See also [Commands](https://minecraft.wiki/w/commands) in the Minecraft Wiki.
+
+Commands are very useful because it provides a way for players to test
+your mod faster and cheat when he doesn't want to spend a long time to do
+not instresting things.
+
+Mojang provides a complete command system in the `com.mojang.brigadier` package, which includes
+ - Command parsing and handling
+ - Primitive [argument types](#argument-types)
+ - [Command Contexts](#command-contexts)
+ - Suggestions
+ - [Exception handling](#exceptions)
+ - Some other small things.
+
+ In fact, Brigadier is the command API of Minecraft.
+
+## Commands architecture
+Here are the main classes and concepts that command creation has.
+
+### Command Dispatcher
+The `CommandDispatcher<S>` (`S` is often `CommandSourceStack` in this case) work like a [`DeferredRegister`](../concepts/registries.md), but it's internally created.
+Its function `register` require a `LiteralArgumentBuilder` argument.
+
+### Argument Builder
+The `ArgumentBuilder` class is abstract, so it can't be used on its own.
+It has two children:
+ - `LiteralArgumentBuilder`: `CommandDispatcher#register` only accepts it. Returned after literal arguments (one-possiblity) and executions
+ - RequiredArgumentBuilder: it has more parameters and associated get functions than `LiteralArgumentBuilder`. Returned after required arguments declarations.
+
+In fact, the entire commands are built with it.
+Last but not least, command trees are chains of builders created with some functions:
+ - `#then(ArgumentBuilder<S, ?>)`: create a new argument.
+ - `#executes(Command)`: execute the command. `Command` can be written like a `Function<CommandContext<CommandSourceStack>, Integer>` where you need to return an integer: `Command.SINGLE_SUCCESS` (or just `1`) for success and `0` otherwise.
```

Comment:
This is still wrong. The return value represents how many time the command performed the desired operation.